parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -145,15 +145,18 @@ public class ParquetRewriter implements Closeable {
   private final Queue<TransParquetFileReader> inputFiles = new LinkedList<>();
   private final Queue<TransParquetFileReader> inputFilesToJoin = new LinkedList<>();
   private final MessageType outSchema;
+  private final MessageType outSchemaWithRenamedColumns;
```

Comment:
Is it possible not to add a new `outSchemaWithRenamedColumns` and stick to `outSchema` with field name replaced?

Reply:
I tried that originally but it just doesn't work. There are places where we need to have old schema. For example when we read the data we need to provide an old path in original schema, if we use a new schema, renamed column just won't be able found and won't be able to read as it is just not there. Maybe we can have some sort of abstraction like a some schema holder that allows to extract both old and rename columns through a single entity but I don't it is possible to avoid the necessity of having both somewhere in the code.

Reply:
Do we support the case where some input files have renamed columns but others don't?

Reply:
As I understand that behavior is not supported by current master branch version of ParquetRewriter and I didn't plan to change it in this PR, it expects the schema for all input files to be the same, both main and join input(if provided), so there are two schemas at max: one schema(if no join files provided), two schemas(if join files are provided). If some of the files are missing a column then I think `ensureSameSchema()` will fail.

Renaming feature doesn't change that, a renamed column can be in the main input or it can be in a join input. In case of join column name overlap with main input, the result of a join depends on a `overwriteInputWithJoinColumns` option, but it doesn't change the output of renaming, if that column is to be renamed it should not change the source of input data being main or join files.

Reply:
> Is it possible not to add a new outSchemaWithRenamedColumns and stick to outSchema with field name replaced?

I replaced renamed schema variable with dynamic extraction of schema with renamed fields where needed, check it out. Didn't want to do it because of potential overhead but it should be very small.

Reply:
Yes, I think this looks better now. Thanks for making this change!